code/game/objects/items/weapons/implants/exile2.dm

```diff
@@ -1,39 +1,125 @@
-/**
-* Exile Implants 2.0
-*
-* Since away missions are fucked, here's an alternative implementation.
-*
-* Instead of confining someone within an away mission, this locks someone to the asteroid.
-*/
+
+//There now it's fun
+
 /obj/item/weapon/implanter/exile
 	name = "implanter-exile"
 
 /obj/item/weapon/implanter/exile/New()
-	src.imp = new /obj/item/weapon/implant/exile( src )
+	src.imp = new /obj/item/weapon/implant/exile(src)
 	..()
 	update()
 	return
 
-
 /obj/item/weapon/implant/exile
 	name = "exile"
-	desc = "Prevents you from returning from the asteroid"
+	desc = "Prevents returning to where you were implanted"
+	var/illegalZ = null
+	var/mob/living/theExile = null
+	var/siteOfImplant = null
+	var/beingDeported = FALSE
+	var/beenSpaced = FALSE
+	var/disablePhrase = ""
+	var/list/zlevels = list(STATION_Z, TELECOMM_Z, DERELICT_Z, ASTEROID_Z, SPACEPIRATE_Z)
+	var/zBanishment_key
 
 /obj/item/weapon/implant/exile/get_data()
 	var/dat = {"
 <b>Implant Specifications:</b><BR>
 <b>Name:</b> Nanotrasen Employee Exile Implant<BR>
-<b>Implant Details:</b> The host of this implant will be prevented from returning to the station."}
+<b>Implant Details:</b> The host of this implant will be prevented from returning to the implant location.<BR>
+<b>For non-permanent use a disable phrase may be assigned on application.<BR>"}
 	return dat
 
-/obj/item/weapon/implant/exile/implanted(mob/M)
-	if(!istype(M, /mob/living/carbon))
-		return 0
-	var/mob/living/carbon/I = M
-	to_chat(I, "<span class='notice'>You shiver as you feel a weak bluespace void surround you.</span>")
-	I.locked_to_z = ASTEROID_Z
+/obj/item/weapon/implant/exile/implanted(mob/source as mob)
+	theExile = source
+	disablePhrase = input("Choose phrase to disable:") as text
```

Comment:
`input`s without a cancel option are annoying.
calling `input` without passing a mob will default to usr, which is bad.
use this instead:
```suggestion
	disablePhrase = stripped_input(user, "Choose phrase to disable:")
```
The phrasing is a bit awkward as well.
And you should now handle the case in which disablePhrase ends up being null, and `return 0` accordingly